packages/web-components/src/pane-switcher/pane-switcher.ts

```diff
@@ -0,0 +1,126 @@
+import { FASTElement } from '@microsoft/fast-element';
+import { isHTMLElement, keyArrowDown, keyArrowUp } from '@microsoft/fast-web-utilities';
+import { Pane } from '../pane/pane.js';
+
+export class PaneSwitcher extends FASTElement {
+  private activePaneIndex: number | null = null;
+  private toggleButtons: HTMLElement[] = [];
+  private panes: Pane[] = [];
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    this.addEventListener('focus', this.handleFocus);
+    this.addEventListener('keydown', this.handleKeyDown);
+    this.toggleButtons = Array.from(this.querySelectorAll('[slot="toggle-buttons"]'));
+    this.setupToggleButtons();
+    this.setupPanes();
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.removeEventListener('focus', this.handleFocus);
+    this.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  constructor() {
+    super();
+    this.activePaneIndex = null;
+  }
+
+  protected setupToggleButtons(): void {
+    const setup = (slot: HTMLSlotElement) => {
+      if (slot.name === 'toggle-buttons') {
+        const buttons = slot.assignedElements();
+
+        const panes = this.querySelectorAll('fluent-pane') as NodeListOf<Pane>; // Add type assertion here
+
+        buttons.forEach((button, index) => {
+          button.addEventListener('click', () => {
+            this.handleToggle(index, panes);
+          });
+        });
+      }
+    };
+
+    // Add the slotchange listener
+    this.addEventListener('slotchange', e => {
+      setup(e.target as HTMLSlotElement);
+    });
+
+    // Run the setup immediately for the existing slots
+    const slots = this.shadowRoot?.querySelectorAll('slot');
+    slots?.forEach(setup);
+  }
+
+  /**
+   * check if the item is a pane
+   */
+  private dialogRoles = ['dialog', 'complementary'];
+  private isPaneElement = (el: HTMLElement | Pane): boolean => {
+    return el instanceof Pane || (isHTMLElement(el) && this.dialogRoles.includes(el.getAttribute('role') as string));
+  };
+
+  protected setupPanes(): void {
+    const slottedPanes = this.querySelectorAll<Pane>('fluent-pane');
+    this.panes = Array.from(slottedPanes);
+
+    /**
+     * Set the context of the Pane
+     * When the pane is a child component of the PaneSwitcher
+     * we want to signal this through an attribute to update styles.
+     */
+    const filteredPaneItems = this.panes?.filter(this.isPaneElement);
```

Comment:
So, to be clear - a pane which has a role of dialog, can have nested panes with the role of dialog?